### PR TITLE
Introduce a wrapper on top of the api client generated from protobuf

### PR DESCRIPTION
**Pull request by ruseinov**
This PR aims at providing a minimal wrapper on top of the client generated from protobuf definitions.

### CODE REVIEW

**Comment by ruseinov** [comment]
@tylertreat FYI

File: src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,575 @@
...
+                    Ok(_) => return client,
+                    _ => continue,
+                }
+            }
+            Err(LiftbridgeError::BrokersUnavailable)?
+        }
+
+        async fn connect(addr: &str) -> Result<ApiClient<Channel>> {
+            let endpoint = Endpoint::try_from(format!("grpc://{}", addr))
+                .map(|e| e.tcp_keepalive(Some(KEEP_ALIVE_DURATION)))
```
[the remark this answers is not in the document]
Comment (ruseinov) [reply src/lib.rs:378]: I have figured that a tcp_keepalive wouldn't hurt to make it cheaper, especially with ssl. 
I still have to figure out the pool cleanup though, I'll add a TODO for this

File: src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,575 @@
...
+
+        // correlation_id sets the identifier used to correlate an ack with the
+        // published Message. If this is not set, the ack will not have a
+        // correlation id.
+        correlation_id: String,
+
+        // ack_policy controls the behavior of Message acks sent by the server. By
+        // default, Liftbridge will send an ack when the stream leader has written
+        // the Message to its write-ahead log.
+        ack_policy: AckPolicy,
```
[the remark this answers is not in the document]
Comment (ruseinov) [reply src/lib.rs:331]: @tylertreat I did this on purpose to reduce the maintenance should anything change as well the implementation time for the initial version. I couldn't figure out the benefit of this as if anything changes - that would reflect on the wrapper types anyway.

File: src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,575 @@
...
+        read_isr_replica: bool,
+    }
+
+    pub struct Subscription {
+        stream: Streaming<Message>,
+    }
+
+    impl Subscription {
+        //TODO: Implement resubscribe on failure
+        pub async fn next(&mut self) -> Result<Option<Message>> {
```
[the remark this answers is not in the document]
Comment (ruseinov) [reply src/lib.rs:222]: @tylertreat This is not needed in Rust, as soon as this gets out of scope everything gets cleaned up automatically.

https://doc.rust-lang.org/book/ch15-03-drop.html

**Comment by ruseinov** [comment]
@tylertreat thanks for the review, I have introduced some TODOs and will to address them very soon. Then we can merge this and move to liftbridge org if that works for you.

**Comment by tylertreat** [comment]
Yep, lgtm. I believe you need to initiate a transfer to @liftbridge-io after merging.

**Comment by ruseinov** [comment]
@tylertreat Yeah, but I need to be added to the org and have permissions ti create a public repo though!

**Comment by tylertreat** [comment]
@ruseinov Invitation sent

**Comment by ruseinov** [comment]
@tylertreat great! if you could also make me an admin for that one repo - that would be awesome.

I’ll probably be working on ci setup soon as well as changing the settings on how branches and prs get merged.

**Comment by tylertreat** [comment]
Done!

**Pull request merged**